### PR TITLE
feat: add `app.agent.mockHttpclient()` for agent

### PR DESCRIPTION
**Pull request by limerickgds**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change
add `app.agent.mockHttpclient()`

### CODE REVIEW

File: app/extend/agent.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  },
+
+  /**
+   * mock csrf
+   * @method App#mockCsrf
+   * @return {App} this
+   * @sice 1.11
+   */
+  mockCsrf() {
+    this.deprecate('[egg-mock] app.agent not support mockCsrf');
```
Comment (dead-horse) [inline app/extend/agent.js]: 去掉 mockCsrf，mockUrllib 和 httpRequest

File: app/extend/agent.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    return this;
+  },
+
+  /**
+   * @see mm#restore
+   * @method App#mockRestore
```
Comment (dead-horse) [inline app/extend/agent.js]: Agent#mockRestore

File: app/extend/agent.js (hunk 1 above)
Comment (limerickgds) [reply app/extend/agent.js]: 已去掉 mockCsrf,httpRequest 。mockUrllib 保留了，这个算 mockHttpclient 别名。暂时保留？

**Comment by fengmk2** [comment]
commit author 修改一下，改成 github 的账号

image

**Comment by limerickgds** [comment]
@fengmk2 已修改

File: app/extend/agent.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+   * @return {Context} this
+   */
+  mockHttpclient(...args) {
+    if (!this._mockHttpclient) {
+      this._mockHttpclient = mockHttpclient(this);
+    }
+    return this._mockHttpclient(...args);
+  },
+
+  mockUrllib(...args) {
```
Comment (dead-horse) [inline app/extend/agent.js]: 去掉 mckUrllib，已经 Deprecate 了的东西不要再新增了

File: test/mock_agent_httpclient.test.js [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+        assert.deepEqual(data, {
+          get: 'mock url get',
+          post: 'mock url post',
+        });
+        done();
+      });
+  });
+
+  it('should deprecate mockUrllib', done => {
+    agent.mockUrllib(url, {
```
Comment (dead-horse) [inline test/mock_agent_httpclient.test.js]: fix this test case. others LGTM

Comment (limerickgds) [reply test/mock_agent_httpclient.test.js]: sorry，忘记修改了。已经移除，本地 test 通过。

**Pull request merged**